frame/clearing-house/src/lib.rs

```diff
@@ -1707,48 +1730,56 @@ pub mod pallet {
 	impl<T: Config> Pallet<T> {
 		fn check_oracle_guard_rails(
 			market: &Market<T>,
+			oracle_status: &OracleStatus<T>,
 			mark_index_divergence_before: T::Decimal,
 			is_risk_increasing: bool,
 		) -> Result<(), DispatchError> {
 			let was_mark_index_too_divergent =
 				Self::exceeds_max_price_divergence(mark_index_divergence_before);
-			let divergence = Self::mark_index_divergence(market)?;
+			let divergence = Self::mark_index_divergence(market, &oracle_status.price)?;
 			let is_mark_index_too_divergent = Self::exceeds_max_price_divergence(divergence);
 
 			// Block if mark-index divergence was pushed too far
 			ensure!(
-				was_mark_index_too_divergent || !is_mark_index_too_divergent,
+				!oracle_status.is_valid ||
```

Comment:
I didn't get why here and in line `1751` a valid oracle status would throw an error.

Reply:
Another way to frame this check is
```rust
if oracle_status.is_valid && !was_mark_index_too_divergent && is_mark_index_too_divergent {
    return Err(/* ... */)
}
```
But here we use `ensure!`, so we need to invert things and use De Morgan's Laws.

Reply:
I'll add a coment so that future maintainers can understand it